packages/eslint-plugin/lib/rules/harden-exports.js

```diff
@@ -46,23 +46,39 @@ module.exports = {
           /** @type {string[]} */
           let exportNames = [];
           if (exportNode.declaration) {
-            // @ts-expect-error xxx typedef
-            if (exportNode.declaration.declarations) {
-              // @ts-expect-error xxx typedef
+            if (exportNode.declaration.type === 'VariableDeclaration') {
               for (const declaration of exportNode.declaration.declarations) {
                 if (declaration.id.type === 'ObjectPattern') {
                   for (const prop of declaration.id.properties) {
-                    exportNames.push(prop.key.name);
+                    if (prop.type === 'RestElement') {
+                      console.warn('Rest elements are not supported');
+                      continue;
+                    }
+                    if (prop.value.type === 'Identifier') {
+                      exportNames.push(prop.value.name);
+                    } else if (
+                      prop.value.type === 'AssignmentPattern' &&
+                      prop.value.left.type === 'Identifier'
+                    ) {
+                      exportNames.push(prop.value.left.name);
+                    }
```

Comment:
This is missing the recursive case in which `prop.value.type` is itself an ObjectPattern or ArrayPattern (e.g., `export const { wrapper: { propName } } = objWithWrapper;` and `export const [{ wrapper: { propName: exportName } }] = [objWithWrapper];`). That was true before as well, but if we're making changes anyway then we ought to fix the gap.

(and likewise for the elements of an ArrayPattern below)